spec/single_city_spec.rb

```diff
@@ -0,0 +1,98 @@
+ require 'spec_helper'
+
+ describe 'Open Weather API ' do
+ 
+  before(:all) do
+  @json_data = Generator.new.city
+  @random_city_info = Generator.new.city.generate_random_cities(rand(0..(@json_data.array_length)))
+  @api_city = OpenWeatherIO.new.single_city_service
+  p @api_city.call_from_api(@random_city_info['id'])
+  end
+
+  context 'checking data types from requested information from the API' do 
+    
+    it 'should be a hash for coordinates' do 
+      expect(@api_city.retrieve_coordinates).to be_kind_of(Hash)
+    end
```

Comment:
Inconsistent formatting here with spaces between your tests in some places then none in others